swap/simulations_test.go

```diff
@@ -0,0 +1,779 @@
+		maxCheques := 10
+
+		// the peer object used for sending
+		creditorPeer := debitorSvc.peers[creditor]
+
+		// send maxCheques number of cheques
+		for i := 0; i < maxCheques; i++ {
+			// use a price which will trigger a cheque each time
+			creditorPeer.Send(ctx, &testMsgBigPrice{})
+			// we need to wait a bit in order to give time for the cheque to be processed
+			err = waitForChequeProcessed(creditorSvc)
+			if err != nil {
+				return err
+			}
+		}
+
+		// check balances:
+		b1, err := debitorSvc.swap.loadBalance(creditor)
+		if err != nil {
+			return err
+		}
+		b2, err := creditorSvc.swap.loadBalance(debitor)
+		if err != nil {
+			return err
+		}
+
+		if b1 != -b2 {
+			return fmt.Errorf("Expected symmetric balances, but they are not: %d vs %d", b1, b2)
+		}
+		// check cheques
+		var cheque1, cheque2 *Cheque
+		if cheque1, err = debitorSvc.swap.loadLastSentCheque(creditor); err != nil {
+			return errors.New("expected cheques with creditor, but none found")
+		}
+		if cheque2, err = creditorSvc.swap.loadLastReceivedCheque(debitor); err != nil {
+			return errors.New("expected cheques with debitor, but none found")
+		}
+
+		// both cheques (at issuer and beneficiary) should have same cumulative value
+		if cheque1.CumulativePayout != cheque2.CumulativePayout {
+			return fmt.Errorf("Expected symmetric cheques payout, but they are not: %d vs %d", cheque1.CumulativePayout, cheque2.CumulativePayout)
+		}
+
+		// check also the actual expected amount
+		expectedPayout := uint64(maxCheques * (DefaultPaymentThreshold + 1))
+
+		if cheque2.CumulativePayout != expectedPayout {
+			return fmt.Errorf("Expected %d in cumulative payout, got %d", expectedPayout, cheque1.CumulativePayout)
+		}
+
+		return nil
+
+	})
+
+	if result.Error != nil {
+		t.Fatal(result.Error)
+	}
+
+	log.Info("Simulation ended")
+}
+
+// TestBasicSwapSimulation starts 16 nodes, then in a simple round robin fashion sends messages to each other.
+// Then checks that accounting is ok. It checks the actual amount of balances without any cheques sent,
+// in order to verify that the most basic accounting works.
+func TestBasicSwapSimulation(t *testing.T) {
+	nodeCount := 16
+	// create the shared backend and params
+	params, err := newSharedBackendSwaps(nodeCount)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// cleanup backend
+	defer params.backend.Close()
+
+	// initialize the simulation
+	sim := simulation.NewInProc(newSimServiceMap(params))
+	defer sim.Close()
+
+	log.Info("Initializing")
+
+	ctx, cancelSimRun := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancelSimRun()
+
+	_, err = sim.AddNodesAndConnectFull(nodeCount)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	log.Debug("Wait for all connections to be established")
+	// first check: let's make sure that all nodes have been connected to the others
+	// this is the maximum number of possible connections
+	maxConns := nodeCount * (nodeCount - 1) / 2
+CONNS:
```

Comment:
This loop could be replaced with `simulations.VerifyFull(t, sim.Net, ids)` (see https://github.com/ethersphere/swarm/pull/1755/commits/1c198e735d59f970519ce831689d6a33812f7e59). This one doesn't wait but from what I saw from the code from `AddNodesAndConnectFull` it shouldn't return anyway before all peers are connected or some connection failed.